patches/minecraft/net/minecraft/world/storage/WorldInfo.java.patch

```diff
@@ -8,7 +8,40 @@
  
      protected WorldInfo()
      {
-@@ -745,4 +746,24 @@
+@@ -438,7 +439,9 @@
+ 
+     public void func_176142_i(int p_176142_1_)
+     {
+-        this.field_176157_p = p_176142_1_;
++    	if (!net.minecraftforge.common.ForgeHooks.onWeatherClear(MinecraftServer.func_71276_C().func_71218_a(field_76105_j).field_73059_b.field_73251_h)) {
++    		this.field_176157_p = p_176142_1_;
```

Comment:
It might be nice to be able to set the result based on the value returned from the event. Because otherwise you can't influence the weather except cancelling the event.